eidpsam/src/main/java/uk/ac/ucl/eidp/auth/jaspic/HeaderBasicAuthModule.java

```diff
@@ -0,0 +1,156 @@
+  private static final String BASIC = "Basic";
+  private static final String AUTHORIZATION_HEADER = "Authorization";
+  private static final String ALLOW_NOTSECURE_NAME = "allow.notsecure";
+  private static final String CREDENTIALS_ENCODING_NAME = "credentials.encoding";
+  private static final String DEF_ENC = "UTF-8";
+  private static final String DEF_NOT_SEC = "false";
+  
+  // private static final String MESSAGES = "META-INF/Messages";
+  
+  // private static final ResourceBundle resourceBundle = ResourceBundle.getBundle(MESSAGES);
+  
+  private PasswordValidationCallback passwordValidationCallback;
+  
+  @Override
+  public void initialize(
+          MessagePolicy requestPolicy,
+          MessagePolicy responsePolicy,
+          CallbackHandler handler,
+          Map options
+  ) throws AuthException {
+    this.handler = handler;
+    if (null != options) {
+      this.options = options;
+      if (null != this.options) {
+        allowNotSecure = Boolean.parseBoolean(
+                (String) options.getOrDefault(ALLOW_NOTSECURE_NAME, DEF_NOT_SEC)
+        );
+        credentialsEncoding = (String) options.getOrDefault(CREDENTIALS_ENCODING_NAME, DEF_ENC);
+      }
+    }
+  }
+
+  @Override
+  public Class[] getSupportedMessageTypes() {
+    return Arrays.copyOf(MESSAGE_TYPES, MESSAGE_TYPES.length);
+  }
+
+  @Override
+  public AuthStatus validateRequest(
+          MessageInfo messageInfo,
+          Subject clientSubject,
+          Subject serviceSubject
+  ) throws AuthException {
+    final HttpServletRequest req = (HttpServletRequest) messageInfo.getRequestMessage();
+    
+    // Evaluate whether not secure connections are allowed
+    if (!req.isSecure() && !allowNotSecure) {
+      return AuthStatus.SEND_FAILURE;
+    }
+    
+    final String userName = readAuthenticationHeader(messageInfo, clientSubject);
+
+    if (null == userName) {
+      return AuthStatus.FAILURE;
+    }
+
+    return AuthStatus.SUCCESS;
+
+  }
+
+  @Override
+  public AuthStatus secureResponse(MessageInfo messageInfo, Subject subject) throws AuthException {
+    return AuthStatus.SEND_SUCCESS;
+  }
+
+  @Override
+  public void cleanSubject(MessageInfo messageInfo, Subject subject) throws AuthException {
+    if (null != subject) {
+      subject.getPrincipals().clear();
+    }
+  }
+  
+  protected void setPasswordValidationCallback(PasswordValidationCallback pwdValidationCallback) {
+    this.passwordValidationCallback = pwdValidationCallback;
+  }
+  
+  private String readAuthenticationHeader(MessageInfo msgInfo, Subject subj) throws AuthException {
+
+    HttpServletRequest request = (HttpServletRequest) msgInfo.getRequestMessage();
+
+    String header = request.getHeader(AUTHORIZATION_HEADER);
+
+    if (!isEmpty(header) && header.startsWith(BASIC + " ")) {
+
+      header = header.substring(6).trim();
+
+      // Decode and parse the authorization header
+      byte[] decBytes = Base64.getDecoder().decode(header.getBytes());
```

Comment:
\[FindBugs\] WARNING: Dm: Found reliance on default encoding in uk\.ac\.ucl\.eidp\.auth\.jaspic\.HeaderBasicAuthModule\.readAuthenticationHeader\(MessageInfo, Subject\): String\.getBytes\(\)